v5.2.2: Update TLS docs (#6323)
pingcap/docs: enable-tls-between-clients-and-servers.md

The TiDB server supports the encrypted connection based on the TLS (Transport Layer Security). The protocol is consistent with MySQL encrypted connections and is directly supported by existing MySQL clients such as MySQL Client, MySQL Shell and MySQL drivers. TLS is sometimes referred to as SSL (Secure Sockets Layer). Because the SSL protocol has [known security vulnerabilities](https://en.wikipedia.org/wiki/Transport_Layer_Security), TiDB does not support SSL. TiDB supports the following protocols: TLS 1.0, TLS 1.1, TLS 1.2 and TLS 1.3.

When using an encrypted connection, the connection has the following security properties:

- Confidentiality: the traffic plaintext is encrypted to avoid eavesdropping
- Integrity: the traffic plaintext cannot be tampered
- Authentication: (optional) the client can verify the identity of the server and the server can verify the identity of the client avoid man-in-the-middle attacks

> To use connectecions secured with TLS you must first configure the TiDB server to enable TLS, then you need to configure the client application to use TLS. Most client applications enable TLS automatically when the server supports this.

Reviewer: Spelling error. Also: 
- Is it a property of the client application, the client, or the client library?
- Is it based on the server supporting TLS or having it configured correctly?
```suggestion
To use connections secured with TLS you must first configure the TiDB server to enable TLS, then you need to configure the client application to use TLS. Most client libraries enable TLS automatically when the server has TLS support configured correctly.
```

>
> If the login user has configured using the [TiDB Certificate-Based Authentication for Login](/certificate-authentication.md#configure-the-user-certificate-information-for-login-verification), the user is implicitly required to enable the encrypted connection to TiDB.

## Configure TiDB to use secure connections

See the following desrciptions about the related parameters to enable secure connections:

- [`auto-tls`](/tidb-configuration-file.md#auto-tls): Enable automatic certificate generation
- [`ssl-cert`](/tidb-configuration-file.md#ssl-cert): specifies the file path of the SSL certificate
- [`ssl-key`](/tidb-configuration-file.md#ssl-key): specifies the private key that matches the certificate
- [`ssl-ca`](/tidb-configuration-file.md#ssl-ca): (optional) specifies the file path of the trusted CA certificate

> Enabling `auto-tls` is enough to get support for secure connections. If validation of client certifictates is needed or if you need more control on how the certificates are generated then using the `ssl-cert`, `ssl-key` and `ssl-ca` variables need to be set as described below.

Author: Thanks. Just note that `auto-tls` does make it possible to use secure connections, but it doesn't really 'ensure' this like `require-secure-transport` does.
